### PR TITLE
Create csv 2

### PR DESCRIPTION
**Pull request by emilyb7**
Script that takes all json files in `/locales` and transforms them to a `.csv` formatted file which we can send to a translator.
Related #137 

To test this:
- make sure you have files for en and chn 
- run `node lib/parse_json_csv.js`

### CODE REVIEW

**Comment by njsfield** [comment]
You mention in [the related issue](https://github.com/InFact-coop/YiMovi/issues/137);

>To make this simpler:
can't have multiline translation strings
no nesting of json more than 1 object deep
no arrays
As a result of which:
need to fix keys for flash messages
create new model for team members with english and chinese names

My pr [here](https://github.com/InFact-coop/YiMovi/pull/153) is a working feature without the shallow-json-restriction outlined here. The changes you've put here will lead to some breaking changes for my PR.

*Please* communicate to me what you'd actually like me to do here

If you'd like me to review this for the functionality only, just assign me in the issue to check out the branch and not raise a PR (if we agree on the functionality, then my [PR](https://github.com/InFact-coop/YiMovi/pull/153) can get merged, then I can raise another PR to restructure JSON and this can get reviewed and merged)

If you'd like me to make changes in my PR to reflect this json structure, let me know.

**Comment by njsfield** [comment]
This fails if translation_csv directory does not exist. So you can add the lines at the bottom of the parse_json_csv file;

```javascript
...

if (!fs.existsSync('./translation_csv')){
  fs.mkdirSync('./translation_csv');
}

createCsvFile(csvString);
```

You check out the branch **locale_routing_pages_flat_json** which is branched off my [pr](https://github.com/InFact-coop/YiMovi/pull/153), I've done the following;
- Flattened JSON structure
- Pulled this branch in (and added the above code for checking directory exists
- Moved team member names temporarily to static array in about partial (no model change yet)

**Comment by emilyb7** [comment]
@njsfield I know about the issues with the json. Your branch with the json files is already merged into this one and I've fixed the json on this branch only. There wasn't much nested stuff, it's a quick change.

For the team members in the array, I suggest we hardcode these in the template for now.

**Comment by emilyb7** [comment]
In  any case, your PR needs to be finished first. It's far more urgent. I Can fix anything else at the end of the project.

**Comment by njsfield** [comment]
Mr branch is not merged into this one, this only contains 5 changes, primarily the JSON files themselves.

I mentioned above my **locale_routing_pages_flat_json** branch, and the changes made in it.

Do you propose I pull those changes into [my open pr](https://github.com/InFact-coop/YiMovi/pull/153) ?

**Comment by emilyb7** [comment]
@njsfield no don't worry about it. Getting this PR merged is not urgent at this stage. #153 is much more important.

**Comment by emilyb7** [comment]
And oh yeah, I started off with your branch merged in, so the 2 sets of features have been tested together. I separated out the commits so that it would be easier to review this PR. The json in this branch has been flattened though.

File: lib/parse_json_csv.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const fs = require('fs');
+const _ = require('underscore');
+const { locales, } = require('../routes/setup/locales.js');
+
+const pathToJson = locale => `./locales/${locale}.json`;
```
Comment (des-des) [inline lib/parse_json_csv.js]: path.join?

File: lib/parse_json_csv.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+const getFileData = locale =>
+  JSON.parse(fs.readFileSync(pathToJson(locale)));
+const flattenArrKeys = arr => _.flatten(_.keys(arr)
+  .map(key => _.keys(arr[key])
+    .map(innerKey => `${key}.${innerKey}`)
```
Comment (des-des) [inline lib/parse_json_csv.js]: indentation?

**Review by des-des — CHANGES_REQUESTED** [review]

**Review by des-des — APPROVED** [review]

**Comment by des-des** [comment]
@emilyb7 conflicts

**Comment by emilyb7** [comment]
@des-des fixed!

**Pull request merged**